File: frontend/lib/models/service_model.dart
class Service {
  final String id;
  final String name;
  final bool hasFee;
  final double fee;
  final bool isPaused;
  final bool allowUrgent;

  Service({
    required this.id,
    required this.name,
    required this.hasFee,
    required this.fee,
    required this.isPaused,
    required this.allowUrgent,
  });

  factory Service.fromJson(Map<String, dynamic> json) {
    return Service(
      id: json['_id'],
      name: json['name'],
      hasFee: json['hasFee'] ?? false,
      fee: (json['fee'] ?? 0).toDouble(),
      isPaused: json['isPaused'] ?? false,
      allowUrgent: json['allowUrgent'] ?? false,
    );
  }
}

File: backend/src/models/serviceModel.js
import mongoose from "mongoose";

const serviceSchema = new mongoose.Schema({
  name: String,
  departmentId: { type: mongoose.Schema.Types.ObjectId, ref: "Department" },
  counters: [{ type: mongoose.Schema.Types.ObjectId, ref: "Counter" }],
  hasFee: { type: Boolean, default: false },
  fee: { type: Number, default: 0 },
  serviceType: {
    type: String,
    enum: ["Documents", "Fees"],
    required: true,
  },
  isPaused: { type: Boolean, default: false },
  allowUrgent: { type: Boolean, default: false },
});

export default mongoose.model("Service", serviceSchema);
